### ISSUE TITLE
ndots is larger than it needs to be

### ISSUE THREAD

**Issue by dchambers** [rancher/rancher#8172]
Howdy!

If I run `cat /etc/resolv.conf` I get this:

```
search default.svc.cluster.local svc.cluster.local cluster.local kubelet.kubernetes.rancher.internal kubernetes.rancher.internal rancher.internal
nameserver 10.43.0.10
options ndots:5
```

where `ndots` is set to `5`.

However, based on this Kubenetes documentation:

> v1.2 introduces a beta feature where the user can specify a Pod annotation, pod.beta.kubernetes.io/subdomain, to specify the Pod’s subdomain. The final domain will be “ ...svc.". For example, a Pod with the hostname annotation set to "foo", and the subdomain annotation set to "bar", in namespace "my-namespace", will have the FQDN "foo.bar.my-namespace.svc.cluster.local"

we can see that the longest possible FQDN is something like `foo.bar.my-namespace.svc.cluster.local`.

Given that the smallest search name is `cluster.local`, this means that the biggest relative query that could be issued is something like `foo.bar.my-namespace.svc`, which only has 3 dots, meaning that the optimal size for `ndots` here is actually `4`.

Although this is only a minor performance tweak (it avoids unnecessary strain on the local DNS and reduces the latency of clients performing absolute name look-ups), it also happens that Alpine Linux currently has a bug that seems to affect some clusters, that prevents absolute look-ups from working at all once relative look-ups have been tried first. You can see issues relating to this both [here](https://github.com/gliderlabs/docker-alpine/issues/8#issuecomment-223901519) and [here](https://github.com/rancher/rancher/issues/5832).

I've raised a bug on the musl libc mailing list (the version of libc that Alpine Linux uses) to hopefully get this fixed, but this would seem like an even easier 'fix' (improvement at least) on the Rancher side since it just involves changing a single variable, which has a positive performance impact too.

Please let me know what you think?

**Comment by dchambers** [comment]
It looks like the underlying problem that inspired me to create this issue [may be a Rancher bug too](https://github.com/rancher/rancher/issues/8180).

Thanks!

**Comment by YoranSys** [comment]
Hi,
With kubernetes and docker running Alpine Linux without cropping manually  the ````search```` field 
I can't resolve name.

Before (not working) : 

````
search default.svc.cluster.local svc.cluster.local cluster.local kubelet.kubernetes.rancher.internal kubernetes.rancher.internal rancher.internal
nameserver 10.43.0.10
options ndots:5
````


````BASH
$ nslookup google.com
nslookup: can't resolve '(null)': Name does not resolve
nslookup: can't resolve 'google.fr': Name does not resolve
````
After cropping (working):

````
search default.svc.cluster.local svc.cluster.local cluster.local
nameserver 10.43.0.10
options ndots:5
````

````BASH
$ nslookup google.com
nslookup: can't resolve '(null)': Name does not resolve

Name:      google.fr
Address 1: 74.125.133.94 wo-in-f94.1e100.net
Address 2: 2a00:1450:400c:c07::5e wo-in-x5e.1e100.net
````

Do you experience same problem ?


I use image like docker:dind based on alpine.

**Comment by timwebster9** [comment]
I'm currently experiencing something that might be related to this.  I'll describe below and I can raise a new issue if it turns out to be something else.

I have a Kubernetes cluster running in Rancher on 3 bare-metal nodes.  For some reason containers in pods I've deployed via `kubectl` can't resolve internet addresses properly.  Commands like `nslookup`, `curl`, `git clone` and anything http-related all fail.  Any internet address resolves to 192.168.1.116, which is the address of my internal DNS server (which is configured on the hosts):

    root@buildit-jenkins-7b7587bcbd-tf8sg:/# ping github.com
    PING neptune.webster.com (192.168.1.116): 56 data bytes
    64 bytes from 192.168.1.116: icmp_seq=0 ttl=63 time=0.368 ms


Other commands like dig and drill (on Alpine) work properly.  This behavior occurs in both Debian and Alpine images.

If I comment out this line in /etc/resolv.conf then everything starts working:

    options ndots:5

So I'm not sure what's going on at all here.  Anyone have any ideas?  Here's my resolv.conf:

    nameserver 10.43.0.10
    search default.svc.cluster.local svc.cluster.local cluster.local rancher.internal rancher.zone  webster.com
    options ndots:5

**Comment by loganhz** [comment]
Thanks for your report!

With the release of Rancher 2.0, development on v1.6 is only limited to critical bug fixes and security patches.

If you think we should keep this issue open, please let me know.
